rename.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+import tempfile
+import textwrap
+
+TEMPLATE_SNAKE = 'pulp_plugin_template'
+TEMPLATE_CAMEL = 'PulpPluginTemplate'
+TEMPLATE_CAMEL_SHORT = 'PluginTemplate'
+TEMPLATE_DASH = 'pulp-plugin-template'
+TEMPLATE_DASH_SHORT = 'plugin-template'
+IGNORE_DIRS = ['.git']
+IGNORE_FILES = ['.gitignore', 'LICENSE', 'rename.py', 'flake8.cfg']
+
+
+def is_valid(name):
+    """
+    Check if specified name is compliant with requirements for it
+    """
+    return bool(re.match(r'^[a-z][0-9a-z_]{2,15}$', name))
+
+
+def to_camel(name):
+    """
+    Convert plugin name from snake to camel case
+    """
+    return name.title().replace('_', '')
+
+
+def to_dash(name):
+    """
+    Convert plugin name from snake case to dash representation
+    """
+    return name.replace('_', '-')
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                     description='rename template data to a specified plugin name')
+    parser.add_argument('plugin_name', type=str,
+                        help=textwrap.dedent('''\
+                            set plugin name to this one
+
+                            Requirements for plugin name:
+                             - specified in the snake form: your_new_plugin_name
+                             - consists of 3 characters or more
+                             - possible characters are letters [a-z], numbers [0-9], underscore [_]
+                             - first character should be a letter [a-z]
+                        '''))
+    args = parser.parse_args()
+    plugin_name = args.plugin_name
+
+    if not is_valid(plugin_name):
+        parser.print_help()
+        return 2
+
+    pulp_plugin_name = 'pulp_' + plugin_name
+    replace_map = {TEMPLATE_SNAKE: pulp_plugin_name,
+                   TEMPLATE_DASH_SHORT: to_dash(plugin_name),
+                   TEMPLATE_DASH: to_dash(pulp_plugin_name),
+                   TEMPLATE_CAMEL_SHORT: to_camel(plugin_name),
+                   TEMPLATE_CAMEL: to_camel(pulp_plugin_name)}
+
+    root_dir = os.path.dirname(os.path.abspath(parser.prog))
+
+    # rename python package directory
+    listed_dir = os.listdir(root_dir)
+    if TEMPLATE_SNAKE in listed_dir:
+        os.rename(os.path.join(root_dir, TEMPLATE_SNAKE),
+                  os.path.join(root_dir, pulp_plugin_name))
+
+    # replace text
+    for dir_path, dirs, files in os.walk(root_dir):
```

Comment:
If we use a copy/rename workflow, it would be good to ignore hidden files that start with "."

Reply:
For now the assumption is that plugin writer doesn't do anything (except maybe just checking if template kinda works right away) before renaming, so only git-related stuff is hidden. It is ignored explicitly. Sounds ok?